### PR TITLE
Static flow control

### PR DESCRIPTION
**Pull request by ElBabbler**
Returning promise on component emit event.

### CODE REVIEW

File: lib/amqp.js [hunk 1]
```diff
@@ -112,7 +112,12 @@ class Amqp {
             log.error('WARNING - buffer full when publishing a message to '
                 + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
         }
-        return result ? result : eventToPromise(this.publishChannel, 'drain').then(() => true);
+
+        if (this.settings.PROCESS_AMQP_DRAIN) {
```
Comment (zubairov) [inline lib/amqp.js]: @ElBabbler please do not use `drain` event but just switch to [ConfirmChannel](http://www.squaremobius.net/amqp.node/channel_api.html#confirmchannel)

Comment (ElBabbler) [reply lib/amqp.js]: But, as I understud, ConfirmChannel can resolve 'acked' or 'nacked' message state only when it consumed, so It takes too much time, and requires sailor working?

File: lib/amqp.js [hunk 2]
```diff
@@ -114,6 +114,13 @@ class Amqp {
                 + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
         }
 
+        try {
+            await this.publishChannel.waitForConfirms();
+        } catch (error) {
+            log.error('Failed on publishing message to queue!!!');
+            throw new Error('Error failed on publishing to MQ' + error);
```
Comment (zubairov) [inline lib/amqp.js]: Why do that? Exception will be propagated higher as it can't be handled on this level IMHO.

File: lib/logging.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ let level;
 if (process.env.NODE_ENV === 'test') {
     level = bunyan.FATAL + 1; //turn off logging
 } else {
-    level = process.env.LOG_LEVEL || 'info';
```
Comment (zubairov) [inline lib/logging.js]: @ElBabbler that would help, apart from that I think 'info' was wrongly used in other pieces of code for `salior`, see here - https://github.com/elasticio/sailor-nodejs/pull/81/commits/2d6bf73746bb3b85456b304f9e9074d2465c57d6

Comment (ElBabbler) [reply lib/logging.js]: Done!

Comment (kirill-levitskiy) [reply lib/logging.js]: @zubairov @ElBabbler can you confirm that everything has been fixed and ready to merge?

**Pull request merged**